lib/components/base-components/Renderable.ts

```diff
@@ -18,6 +18,7 @@ export const orderedRenderPhases = [
   "CreateTraceHintsFromProps",
   "SourceGroupRender",
   "AssignNameToUnnamedComponents",
+  "SourceNameValidation",
```

Comment:
If this renderPhase is just for explicitly named components, and not auto assigned names. I think it's better you do it before the `AssignNameToUnnamedComponents` phase

Reply:
this should be immediately after ReactSubtreesRender i think, and it's not `SourceNameValidation` i would call it `SourceNameDuplicateComponentRemoval`